maat-court-data-api/src/main/java/gov/uk/courtdata/hardship/impl/HardshipReviewImpl.java

```diff
@@ -0,0 +1,85 @@
+package gov.uk.courtdata.hardship.impl;
+
+import gov.uk.courtdata.dto.HardshipReviewDTO;
+import gov.uk.courtdata.entity.HardshipReviewDetailEntity;
+import gov.uk.courtdata.entity.HardshipReviewEntity;
+import gov.uk.courtdata.entity.HardshipReviewProgressEntity;
+import gov.uk.courtdata.hardship.mapper.HardshipReviewMapper;
+import gov.uk.courtdata.model.hardship.HardshipReviewDetail;
+import gov.uk.courtdata.model.hardship.HardshipReviewProgress;
+import gov.uk.courtdata.repository.HardshipReviewDetailReasonRepository;
+import gov.uk.courtdata.repository.HardshipReviewRepository;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Component;
+
+import java.util.List;
+
+@Component
+@RequiredArgsConstructor
+@Slf4j
+public class HardshipReviewImpl {
+
```

Comment:
I would add some loggers and try/catch in every method

Reply:
All the exceptions in are being thrown up until they are handled in the RestControllerAdviser class, where the logging happens. I have tried to clean that class up, set correct HTTP status codes and updated to code to throw a more appropriate business exception rather than something meant to be used in the Collections API. However, there are couple of pre-existing exception handlers in there which set an inappropriate HTTP status code so I had to leave them alone to avoid breaking the contract with existing consumers